src/openloco/ui.cpp

```diff
@@ -760,22 +766,118 @@ namespace openloco::ui
         SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, title.c_str(), message.c_str(), window);
     }
 
-    bool updateSDLResolution()
+    config::resolution_t* getDisplayResolutionByMode(config::screen_mode mode, int32_t displayIndex)
     {
-        SDL_DisplayMode mode;
-        SDL_GetWindowDisplayMode(window, &mode);
+        config::resolution_t defaultResolution = { 640, 480 };
+        config::resolution_t* outputResolution = &defaultResolution;
 
-        auto& cfg = config::get();
-        mode.h = cfg.resolution_height;
-        mode.w = cfg.resolution_width;
+        auto& config = config::get_new();
+        switch (mode)
+        {
+            case config::screen_mode::window:
+                if (config.display.window_resolution.width > 0 && config.display.window_resolution.height > 0)
+                    outputResolution = &config.display.window_resolution;
+                else
+                    outputResolution = &defaultResolution;
+                break;
+            case config::screen_mode::fullscreen:
+                if (config.display.fullscreen_resolution.width > 0 && config.display.fullscreen_resolution.height > 0)
+                    outputResolution = &config.display.fullscreen_resolution;
+                break;
+            case config::screen_mode::fullscreen_borderless:
+                break;
+            default:
+                console::error("Unrecongised display mode");
+                break;
+        }
 
-        if (SDL_SetWindowDisplayMode(window, &mode) != 0)
+        // If we have no resolution in config, or it's 0x0, use the current desktop resolution instead
+        if (outputResolution == nullptr)
+        {
+            config::resolution_t desktopResolution = getDesktopResolution(displayIndex);
+            outputResolution = &desktopResolution;
+        }
+        else if (!(outputResolution->width > 0 && outputResolution->height > 0))
+        {
+            config::resolution_t desktopResolution = getDesktopResolution(displayIndex);
+            outputResolution = &desktopResolution;
+        }
```

Comment:
Both of these branches have the same body.
```suggestion
        if (outputResolution == nullptr || !(outputResolution->width > 0 && outputResolution->height > 0))
        {
            config::resolution_t desktopResolution = getDesktopResolution(displayIndex);
            outputResolution = &desktopResolution;
        }
```